test-suite/src/test/scala/org/scalajs/testsuite/javalib/CollectionsTest.scala

```diff
@@ -0,0 +1,1332 @@
+    }
+  }
+
+  for (factory <- SetFactory.allFactories) {
+    describe(s"java.util.Collections.synchronizedSet[T](${factory.implementationName}[T]): Set[T]") {
+      testSetApi(
+        new SetFactory {
+          def implementationName: String =
+            s"synchronizedSet(${factory.implementationName})"
+
+          def empty[E]: ju.Set[E] =
+            ju.Collections.synchronizedSet(factory.empty[E])
+
+          def allowsNullElement: Boolean =
+            factory.allowsNullElement
+        }
+      )
+    }
+  }
+
+  for (factory <- SortedSetFactory.allFactories) {
+    describe("java.util.Collections.synchronizedSortedSet[T]" +
+        s"(${factory.implementationName}[T]): SortedSet[T]") {
+      testSortedSetApi(
+        new SortedSetFactory {
+          def implementationName: String =
+            s"synchronizedSortedSet(${factory.implementationName})"
+
+          def empty[E]: ju.SortedSet[E] =
+            ju.Collections.synchronizedSortedSet(factory.empty[E])
+
+          def allowsNullElement: Boolean =
+            factory.allowsNullElement
+        }
+      )
+    }
+  }
+
+  for (factory <- ListFactory.allFactories) {
+    describe("java.util.Collections.synchronizedList[T]" +
+        s"(${factory.implementationName}[T]): List[T]") {
+      testListApi(
+        new ListFactory {
+          def implementationName: String =
+            s"synchronizedList(${factory.implementationName})"
+
+          def empty[E]: ju.List[E] =
+            ju.Collections.synchronizedList(factory.empty[E])
+        }
+      )
+    }
+  }
+
+  for (factory <- MapFactory.allFactories) {
+    describe("java.util.Collections.synchronizedMap[K, V]" +
+        s"(${factory.implementationName}[K, V]): Map[K, V]") {
+      testMapApi(
+        new MapFactory {
+          def implementationName: String =
+            s"synchronizedMap(${factory.implementationName})"
+
+          def empty[K, V]: ju.Map[K, V] =
+            ju.Collections.synchronizedMap(factory.empty[K, V])
+
+          override def allowsNullKeys: Boolean =
+            factory.allowsNullKeys
+
+          override def allowsNullValues: Boolean =
+            factory.allowsNullValues
+        }
+      )
+    }
+  }
+
+  for (factory <- SortedMapFactory.allFactories) {
+    describe("java.util.Collections.synchronizedSortedMap[K, V]" +
+        s"(${factory.implementationName}[K, V]): SortedMap[K, V]") {
+      testSortedMapApi(
+        new SortedMapFactory {
+          def implementationName: String =
+            s"synchronizedSortedMap(${factory.implementationName})"
+
+          def empty[K, V]: ju.SortedMap[K, V] =
+            ju.Collections.synchronizedSortedMap(factory.empty[K, V])
+
+          override def allowsNullKeys: Boolean =
+            factory.allowsNullKeys
+
+          override def allowsNullValues: Boolean =
+            factory.allowsNullValues
+        }
+      )
+    }
+  }
+
+  class A extends jl.Comparable[A] {
+    def compareTo(o: A): Int = ## - o.##
```

Comment:
Please write `this.##` instead of just `##`. Seeing just `##` is very disturbing.